src/components/checkbox/checkbox/checkbox.test.tsx

```diff
@@ -0,0 +1,45 @@
+import { render, screen } from '@testing-library/react';
+import { IconType } from '../../icon';
+import { Checkbox, type ICheckboxProps } from './checkbox';
+
+describe('<Checkbox /> component', () => {
+    const createTestComponent = (props?: Partial<ICheckboxProps>) => {
+        const completeProps = {
+            label: 'test-label',
+            ...props,
+        };
+
+        return <Checkbox {...completeProps} />;
+    };
+
+    it('renders a checkbox with the specified label', () => {
+        const label = 'check';
+        render(createTestComponent({ label }));
+        expect(screen.getByRole('checkbox', { name: label })).toBeInTheDocument();
+    });
+
+    it('sets a random id to the form field when the id property is not set', () => {
+        render(createTestComponent());
+        expect(screen.getByRole('checkbox').id).toBeDefined();
+    });
+
+    it('renders the unchecked state when the checkbox is not checked', () => {
+        const checked = false;
+        render(createTestComponent({ checked }));
+        expect(screen.getByTestId(IconType.CHECKBOX_DEFAULT)).toBeInTheDocument();
```

Comment:
Careful with this assertion as it's slightly misleading. The `CHECKBOX_DEFAULT` is technically always in the document, no? It's only *visible* now. That said, I'm assuming testing-library isn't playing well with the data attribute?

Reply:
That's true, on the other hand the `getBy*` utility classes already return null if the element is not visible and need an explicit `hidden: true` parameter to get such elements. Either case, checking `toBeVisible` instead of `toBeInTheDocument` makes more sense semantically 👌🏻